src/Symfony/Component/Validator/Tests/Constraints/EmailValidatorTest.php

```diff
@@ -86,4 +88,28 @@ public function getInvalidEmails()
             array('example@localhost'),
         );
     }
+
+    /**
+     * @dataProvider getHostnamesAndEmailAddresses
+     */
+    public function testHostnameIsProperlyParsed($hostnameAndEmail)
+    {
+        $method = new ReflectionMethod($this->validator, 'extractHostname');
+        $method->setAccessible(true);
```

Comment:
@natechicago You can mock the `checkdnsrr()` function (see #16194 for how we did that for the time-related functions).